### DISCUSSION TITLE
Suggestions about data modeling for a school database

### DISCUSSION THREAD

**Discussion by soleuil** [supabase/supabase#10282] [Questions]
Hello,

If I may, I would like to specify that I am not a proffesional programmer but a regular teacher with some programming knowledge just because I love programming and I stay on learning when I have some time to dedicate to it.

One of the projects that I did some time ago was a React application to manage in our school the students, teachers, grades, and related things. I used Firestore as a database and we have used it succesfully but as soon as I heard about Supabase I knew that I wanted to change that project and move to Supabase. So now I am in the process to do that.

The issue that I am facing is that as they are different kind of models (Sql and Nosql) I am almost sure that I have to change the approach in many aspects. I am used to duplicate data everywhere. For example, everytime a grade was saved, the teacherID saving the grade, the studentID, the subjectID, and all that information had to be in the document. Right now I am beginning to create the "students" table and I am thinking that I have to change the model because of the linking possibility. So I come here to ask for advice or some guidance before I create all the data and then I realize that I have to redo certain things (which most likely will happen).

In the students table thatI just began I have the first doubt that is assaulting me. In the previous project, each student had, besides the personal fields, one array value where all of the subjects were listed, but also a subcollection with a document for each of those subjects, so I could specify teacher associated with the subject and student and so on.

I believe that with Supabase I would have to create different tables: students, teachers, subjects, and create links between them. But how would I link several subjects in one field for the student? I am guessing that the JSON value would be the one to specify several subjects with different teachers for that student, but I don't know if I could link that JSON field with other tables.

It must be obvious that I am completely new in the SQL database and most certainty my questions are redundant and dumb, but I would really appreciate some words and guidance so I take some impulse to create my data model and move on with the project as I am a little stuck at this moment.

Thank you!

**Comment by steve-chavez** [comment]
> I believe that with Supabase I would have to create different tables: students, teachers, subjects, and create links between them. But how would I link several subjects in one field for the student?

You would need a join table for many-to-many relationships, see https://en.wikipedia.org/wiki/Associative_entity

Then you can get nested JSON through `supabase-js`, see https://supabase.com/docs/reference/javascript/select#query-foreign-tables-through-a-join-table
